utils/icinga2_client.go

```diff
@@ -35,6 +36,56 @@ func NewIcinga2Client(address string, username string, password string) *Icinga2
 	}
 }
 
+// getNoBody successfully returns http.NoBody.
+func getNoBody() (io.ReadCloser, error) {
+	return http.NoBody, nil
+}
+
+func (c *Icinga2Client) Do(req *http.Request) (*http.Response, error) {
+	if req.Body == nil && req.GetBody == nil {
+		req.Body = http.NoBody
+		req.GetBody = getNoBody
+	}
+
+	var timer *time.Timer
+	nextBody := req.Body
+	ctx := req.Context()
+
+	for attempt := 1; ; attempt++ {
+		req := req.Clone(ctx)
```

Comment:
Please explain why you clone the request here.

I couldn't find a reason for this in the documentation of the `net/http` package, only the following in the [`RoundTripper` interface documentation](https://pkg.go.dev/net/http#RoundTripper):

> Callers should not mutate or reuse the request until the Response's Body has been closed.

So I hope the opposite is true and it's fine to reuse the request after closing the response body (otherwise, why would it mention this condition instead of saying to never mutate or reuse the request).

So I'd kind of expect doing the following in the innermost `if` below should work just fine and should allow to get rid of the clone and the `nextBody` variable:

```go
_ = response.Body.Close()
req.Body = body
```